File: Medium/43MajorityElement.py
def majorityElement(array):
    count = 0
    answer = None

    for value in array:
        if count == 0:
            answer = value

        if value == answer:
            count += 1
        else:
            count -= 1
    return answer


print(majorityElement([1, 2, 3, 2, 2, 1]))

File: Medium/43MajorityElement.js
function majorityElement(array) {
  // Write your code here.
  let count = 0;
  let answer = null;

  for (let i of array) {
    if (count == 0) {
      answer = i;
    }

    if (answer == i) {
      count++;
    } else count--;
  }
  return answer;
}
